### PR TITLE
chlore: Form Improvements & autocomplete implementation

### PR DESCRIPTION
**Pull request by masskoder**
### Scope of changes

This PR addresses the following issues:

- Introduces a new business category.
- Enhances error handling for file uploads.
- Refines the state management within forms.
- Implement vasp autocomplete feature

### Type of change

- [x] bug fix
- [x] new feature
- [ ] documentation
- [x] other (describe)

### Acceptance criteria

https://www.awesomescreenshot.com/video/20146526?key=30b20e736312a1979683c2c8d58cb848

### Author checklist

- [ ] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [ ]  I have added new test fixtures as needed to support added tests
- [ ]   Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [ ]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.

### CODE REVIEW

File: web/gds-user-ui/src/components/Section/SearchDirectory.tsx [hunk 1]
```diff
@@ -83,15 +98,32 @@ const SearchDirectory: React.FC<TSearchDirectory> = ({
...
+                      rollNavigation
+                      // no found message
+                      noOptionsMessage="No VASP found">
+                      <AutoCompleteInput
+                        variant="filled"
+                        placeholder="Common name or VASP ID"
+                        autoFocus
+                      />
+                      <AutoCompleteList>
+                        {Object.keys(options)?.map((oid: any) => (
```
Comment (daniellemaxwell) [inline web/gds-user-ui/src/components/Section/SearchDirectory.tsx:111]: For my understanding, does `oid` = `options ID`?

File: web/gds-user-ui/src/components/Section/SearchDirectory.tsx [hunk 2]
```diff
@@ -83,15 +98,32 @@ const SearchDirectory: React.FC<TSearchDirectory> = ({
...
+                          <AutoCompleteItem
```
Comment (daniellemaxwell) [inline web/gds-user-ui/src/components/Section/SearchDirectory.tsx:112]: I noticed that more than one value had the hover background when I attempted a search and in the acceptance criteria video. 

Here's a video of what I experienced: https://www.awesomescreenshot.com/video/20164689?key=29b03758f4a12fb65f5bedf10e4f6b1f

File: web/gds-user-ui/src/modules/home/useFetchLookup.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    setError(null);
+  };
+  const handleSearch = (searchQuery: string) => {
+    const query = isValidUuid(searchQuery) ? `uuid=${searchQuery}` : `common_name=${searchQuery}`;
+
+    const fetchLookup = async () => {
+      setIsLoading(true);
+      try {
+        const response = await lookup(query);
+        if (!response.mainnet || !response.testnet) setError('No data found');
```
Comment (daniellemaxwell) [inline web/gds-user-ui/src/modules/home/useFetchLookup.ts:24]:
```suggestion
        if (!response.mainnet || !response.testnet) setError('No data found.');
```

File: web/gds-user-ui/src/modules/home/useFetchLookup.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        setData(data);
+        setSearchString(searchQuery);
+      } catch (e: any) {
+        if (!e?.response?.data?.success) {
+          setError(e?.response?.data?.error);
+        } else {
+          setError('Something went wrong');
```
Comment (daniellemaxwell) [inline web/gds-user-ui/src/modules/home/useFetchLookup.ts:31]:
```suggestion
          setError('Something went wrong.');
```

File: web/gds-user-ui/src/hooks/useUploadFile.tsx [hunk 5]
```diff
@@ -50,11 +86,17 @@ const useUploadFile = () => {
...
-        console.log('[] error validationData', e);
-
-        handleError(e, `[Invalid file], it's missing some required fields : ${e.message}`);
-      } finally {
         setIsFileLoading(false);
+        console.log('[] error validationData', e.message);
+
+        toast({
+          title: 'Invalid file',
+          description: e.message || 'Your json file is invalid',
```
Comment (daniellemaxwell) [inline web/gds-user-ui/src/hooks/useUploadFile.tsx:94]:
```suggestion
          description: e.message || 'Your json file is invalid.',
```

**Review by daniellemaxwell — APPROVED** [review]
A lot of big changes made in this one! I made note of a style issue I noticed when using the `Directory Lookup`. If you could take a look, that would be great, unless you have all ready created a story about it. Great work!

File: web/gds-user-ui/src/components/Section/SearchDirectory.tsx (hunk 1 above)
Comment (masskoder) [reply web/gds-user-ui/src/components/Section/SearchDirectory.tsx:111]: > For my understanding, does `oid` = `options ID`?

Yes the oid is the key of option object.

**Pull request merged**